SDK-1204 Updated docs (#686)
box/box-ios-sdk: docs/usage/groups.md

Get Enterprise Groups
---------------------

> To retrieve information about the groups within the enterprise, call [`client.groups.listForEnterprise(name: String?, offset: Int?, limit: Int?, fields: [String]?)`][get-enterprise-groups]. You can also pass in a `name` paramter to act as a filter. This method will return an iterator object in the completion that you can use to retrieve groups in the enterprise.

Reviewer: ```suggestion
To retrieve information about the groups within the enterprise, call [`client.groups.listForEnterprise(name: String?, offset: Int?, limit: Int?, fields: [String]?)`][get-enterprise-groups]. You can also pass in a `name` paramter to act as a filter. This method will return an iterator object in the completion, which is used to retrieve groups in the enterprise.
```

Get Memberships for Group
-------------------------

> To retrieve information about the memberships in a group, call [`client.groups.listMemberships(groupID: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-group] with the ID of the group to retrieve group memberships for. This method will return an iterator object in the completion that you can use to the retrieve memberships.

Reviewer: ```suggestion
To retrieve information about the memberships in a group, call [`client.groups.listMemberships(groupID: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-group] with the ID of the group to retrieve group memberships for. This method will return an iterator object in the completion, which is used to retrieve the memberships.
```

------------------------

To retrieve information about the group memberships for a given user, call

> [`client.groups.listMembershipsForUser(userId: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-user] with the ID of the user to retreive group memberships for. This method will return an iterator object in the completion that you can use to retrieve the memberships.

Reviewer: ```suggestion
[`client.groups.listMembershipsForUser(userId: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-user] with the ID of the user to retreive group memberships for. This method will return an iterator object in the completion, which is used to retrieve the memberships.
```

Get Collaborations for Group
----------------------------

> To retrieve all group collaborations for a given group, call [`client.groups.listCollaborations(groupId: String, offset: Int?, limit: Int? fields: [String]?)`][get-collaborations-for-group] with the ID of the group to retrieve collaborations for. This method will return an iterator object in the completion that you can use to retrieve the collaborations.

Reviewer: ```suggestion
To retrieve all group collaborations for a given group, call [`client.groups.listCollaborations(groupId: String, offset: Int?, limit: Int? fields: [String]?)`][get-collaborations-for-group] with the ID of the group to retrieve collaborations for. This method will return an iterator object in the completion, which is used to retrieve the collaborations.
```
